tidy3d/web/cli/develop/packaging.py

```diff
@@ -0,0 +1,107 @@
+import pathlib
+
+import click
+from pathlib import Path
+import subprocess
+from .index import develop
+from .utils import echo_and_check_subprocess
+
+output_timing_log = ["python", "-X", "importtime", "-c", "import tidy3d"]
+
+# Runs the import 100 times.
+average_test_import = [
+    "python",
+    str(Path("tests", "test_timing", "import_tidy3d.py")),
+]
+
+timing_commands = {
+    "output_timing_log": output_timing_log,
+    "average_test_import": average_test_import,
+}
+
+
+def benchmark_timing_operations(
+    timing_command: str, in_poetry_environment: bool = True, output_file: str = "import.log"
+):
+    """
+    This function is used to time and benchmark the timing performance of various operations in the codebase. The
+    idea of this functionality is to be able to track the performance of various operations over time and normalise
+    it for different hardware. For example, say we just need to use a certain section of the codebase, we can use
+    this functionality to extract the timing performance of that specific operation and compare it to previous usages.
+
+    Note that this is run within the top level of the tidy3d package. We can write specific files with specific
+    operations in the `tests` section and benchmark them properly using this. This function does not require poetry
+    and can be run anywhere where a tidy3d installation is already implemented. The output file has an extension.
+    """
+    timing_command_list = list()
+    if output_file is None:
+        output_file = timing_command.split("_")[1:] + ".log"
+
+    output_file_path = pathlib.Path(output_file)
+
+    try:
+        output_file_write = open(output_file_path, "w+")
+        pass
+    except FileNotFoundError:
+        raise FileNotFoundError(
+            "The output file path "
+            + str(output_file_path)
+            + " does not exist and cannot be created."
+        )
+
+    if in_poetry_environment:
+        timing_command_list += ["poetry", "run"]
+
+    try:
+        timing_command_list = timing_commands[timing_command].copy()
+    except KeyError:
+        # This has to do with choosing a timing command not available in the dictionary
+        raise KeyError(
+            "Make sure the selected timing command "
+            + str(timing_command)
+            + "corresponds to an existing "
```

Comment:
could probably use single quotes around `timing_command` and a space after. (or could use an `f` string).

Reply:
Sorted